### PR TITLE
External HBase Table Integration as Source and Sink.

### PR DESCRIPTION
**Pull request by nitinmotgi**
This PR is not yet ready for review.
- [x] External HBase Source
- [x] External HBase Sink
- [ ] Issue with HBase Classloading on Distributed Cluster (https://issues.cask.co/browse/CDAP-4112).

Build : http://builds.cask.co/browse/CPG-PLUG3-1

### CODE REVIEW

File: hbase-plugins/src/test/java/co/cask/plugin/etl/test/HBaseTest.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
+package co.cask.plugin.etl.test;
```
Comment (rohitsinha54) [inline hbase-plugins/src/test/java/co/cask/plugin/etl/test/HBaseTest.java]: Copyright

File: hbase-plugins/src/main/java/co/cask/plugin/etl/batch/source/HBaseSource.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package co.cask.plugin.etl.batch.source;
```
Comment (rohitsinha54) [inline hbase-plugins/src/main/java/co/cask/plugin/etl/batch/source/HBaseSource.java]: Copyright

File: hbase-plugins/src/main/java/co/cask/plugin/etl/batch/sink/HBaseSink.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
+package co.cask.plugin.etl.batch.sink;
```
Comment (rohitsinha54) [inline hbase-plugins/src/main/java/co/cask/plugin/etl/batch/sink/HBaseSink.java]: Copyright

File: hbase-plugins/src/main/java/co/cask/plugin/etl/batch/HBaseConfig.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package co.cask.plugin.etl.batch;
```
Comment (rohitsinha54) [inline hbase-plugins/src/main/java/co/cask/plugin/etl/batch/HBaseConfig.java]: Copyright

File: hbase-plugins/src/main/java/co/cask/plugin/etl/batch/sink/HBaseSink.java [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import org.apache.hadoop.hbase.mapreduce.TableOutputFormat;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.util.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ *
```
Comment (sreevatsanraman) [inline hbase-plugins/src/main/java/co/cask/plugin/etl/batch/sink/HBaseSink.java]: Comments

**Comment by sreevatsanraman** [comment]
PR is missing UI json for source and sink. Please add that and fix conflicts.

File: pom.xml [hunk 6]
```diff
@@ -48,7 +49,7 @@
 
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <cdap.version>3.2.1</cdap.version>
```
Comment (sreevatsanraman) [inline pom.xml:51]: Let's not make this change. Let's depend on 3.2.1 and make the relevant changes we need to the plugin.

**Comment by gokulavasan** [comment]
@sreevatsanraman Changed to depend on 3.2.1. Please review.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**